Sources/SwiftFrameCore/Extensions/NSError+Extensions.swift

```diff
@@ -2,8 +2,28 @@ import Foundation
 
 public extension NSError {
 
-    convenience init(code: Int = 1, description: String) {
-        self.init(domain: "com.kayak.SwiftFrame", code: code, userInfo: [NSLocalizedDescriptionKey: description])
+    static let kExpectationKey = "kExpectationKey"
+    static let kActualValueKey = "kActualValueKey"
+
+    convenience init(code: Int = 1, description: String, expectation: String? = nil, actualValue: String? = nil) {
+        self.init(
+            domain: "com.kayak.SwiftFrame",
+            code: code,
+            userInfo: [
+                NSLocalizedDescriptionKey: description,
+                NSError.kExpectationKey: expectation as Any,
+                NSError.kActualValueKey: actualValue as Any
+            ])
+    }
+
+    // Not too sure about this naming yet
```

Comment:
Naming seems ok-ish to me. I'd maybe make it `var expectation` to match the `kExpectationKey`.